src/types/tcb_info.rs

```diff
@@ -0,0 +1,399 @@
+
+        let tcb_info: TcbInfo =
+            serde_json::from_str(self.tcb_info_raw.get()).context("tcb info")?;
+
+        if tcb_info
+            .tcb_levels
+            .iter()
+            .any(|e| e.tcb.version() != tcb_info.version)
+        {
+            bail!(
+                "mismatched tcb info versions, should all be {:?}",
+                tcb_info.version,
+            );
+        }
+
+        // tcb_type determines how to compare tcb level
+        // currently, only 0 is valid
+        if tcb_info.tcb_type != 0 {
+            bail!("unsupported tcb type {}", tcb_info.tcb_type,);
+        }
+        Ok(tcb_info)
+    }
+
+    pub fn get_tcb_info(&self) -> anyhow::Result<TcbInfo> {
+        serde_json::from_str(self.tcb_info_raw.get())
+            .map_err(|e| anyhow::anyhow!("tcb info parsing failed: {}", e))
+    }
+}
+
+/// Version of the TcbInfo JSON structure
+///
+/// In the PCS V3 API the TcbInfo version is V2, in the PCS V4 API the TcbInfo
+/// version is V3. The V3 API includes advisoryIDs and changes the format of
+/// the TcbLevel
+
+#[derive(Deserialize, Serialize, Clone, Debug, Eq, PartialEq)]
+#[serde(try_from = "u16")]
+pub(crate) enum TcbInfoVersion {
+    V2 = 2,
+    V3 = 3,
+}
+
+impl TryFrom<u16> for TcbInfoVersion {
+    type Error = &'static str;
+    fn try_from(value: u16) -> std::result::Result<Self, Self::Error> {
+        match value {
+            2 => Ok(TcbInfoVersion::V2),
+            3 => Ok(TcbInfoVersion::V3),
+            _ => Err("Unsupported TCB Info version"),
+        }
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TcbInfo {
+    #[serde(skip_serializing_if = "Option::is_none", rename = "id")]
+    id: Option<String>,
+    version: TcbInfoVersion,
+    _issue_date: chrono::DateTime<Utc>,
+    pub next_update: chrono::DateTime<Utc>,
+    #[serde(with = "hex")]
+    pub fmspc: [u8; 6],
+    #[serde(with = "hex")]
+    pub pce_id: [u8; 2],
+    tcb_type: u16,
+    _tcb_evaluation_data_number: u16,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    tdx_module: Option<TdxModule>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    tdx_module_identities: Option<Vec<TdxModuleIdentity>>,
+    tcb_levels: Vec<TcbLevel>,
+}
+
+impl TcbInfo {
+    pub fn verify_tdx_module(&self, quote_body: &TdxReportBody) -> anyhow::Result<TcbStatus> {
+        if self.tdx_module.is_none() {
+            return Err(anyhow::anyhow!("no tdx module found in tcb info"));
+        }
+
+        let (tdx_module_isv_svn, tdx_module_version) = (quote_body.tee_tcb_svn[0], quote_body.tee_tcb_svn[1]);
+        let tdx_module_identity_id = format!("TDX_{:02}", tdx_module_version);
+
+        if self.tdx_module_identities.is_none() {
+            return Err(anyhow::anyhow!("no tdx module identities found in tcb info"));
+        }
+
+        let tdx_module_identity = self.tdx_module_identities
```

Comment:
This step is only performed if `tdx_module_version > 0`. 

If `tdx_module_version == 0`, the `mrsigner` and `attributes` values are compared with the ones in `TDX Module` instead of `TDX Module Identity`

Reply:
https://github.com/intel/SGX-TDX-DCAP-QuoteVerificationLibrary/blob/7e5b2a13ca5472de8d97dd7d7024c2ea5af9a6ba/Src/AttestationLibrary/src/Verifiers/QuoteVerifier.cpp#L154-L179

Reply:
Done https://github.com/automata-network/dcap-rs/pull/15/commits/2eb1a05f1fd4ec2727b2fec7637565cacdc2868d